docs/fab_security.rst

```diff
@@ -0,0 +1,63 @@
+..  Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+..    http://www.apache.org/licenses/LICENSE-2.0
+
+..  Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+Airflow UI Security
+===================
+
+Security of Airflow Webserver UI is handled by Flask AppBuilder (FAB).
+Please read its related `security document <http://flask-appbuilder.readthedocs.io/en/latest/security.html>`_
+regarding its security model.
+
+Default Roles
+-------------
+Airflow ships with a set of roles by default: Admin, User, Op, Viewer, and Public.
+Only ``Admin`` users could configure/alter the permissions for other roles. But it is not recommended
+that ``Admin`` users alter these default roles in any way by removing
+or adding permissions to them as these roles will be re-synchronized to their original values.
```

Comment:
@feng-tao , I'm not sure about this line. I remember https://github.com/apache/airflow/pull/4118 fixed it and now permission change (adding/removing) will be persisted.

Please correct me if I'm wrong or have misunderstood something.

Reply:
@XD-DENG  ah, the fix is for roles in ROLE_CONFIGS, which does not include admin, so @feng-tao  is correct in this statement. `update_admin_perm_view()` will add all permissions for admin.

It's a bit restricting, but I think this is a fair restriction to make given it's admin/

Reply:
Thanks @jgao54 for clarification. But it's still not clear to me (please bear with me if I'm misunderstanding anything).

The impression I get from this paragraph was: 1. `Admin` can configure/alter permission for other roles (including `User`, `Op`, `Viewer`, and `Public`); 2. But doing this is not recommended given these roles (`User`, `Op`, `Viewer`, and `Public`) will be re-synchronized to original values, all changes will be gone after a short while.

Let's say I add `can_dagrun_clear` permission to `Viewer` (which was not granted to `Viewer` in original values). And this change will persist since https://github.com/apache/airflow/pull/4118 merged. But the impression I get from this documentation is this change will not persist.

Reply:
@XD-DENG  My bad, I re-read the lines and I agree with you it is confusing. In my mind I'm making up words on the fly as I read it:
"it's not recommended for admin to change **its own permissions**" is what I thought it suggested.

But yes, since #4118, permission change will be persisted for other roles.